pkg/services/accesscontrol/evaluator/evaluator.go

```diff
@@ -39,18 +43,30 @@ func evaluateScope(dbScopes map[string]struct{}, targetScopes ...string) (bool,
 	for _, s := range targetScopes {
 		var match bool
 		for dbScope := range dbScopes {
-			rule, err := glob.Compile(dbScope, ':', '/')
+
+			prefix := dbScope[0 : len(dbScope)-1]
+			if strings.ContainsAny(prefix, "*?") {
+				msg := fmt.Sprintf("Invalid scope: %v should not contain meta-characters like * or ?, except in the last position", dbScope)
+				logger.Error(msg, "scope", dbScope)
+				return false, errors.New(msg)
+			}
+
+			rule, err := glob.Compile(dbScope)
 			if err != nil {
 				return false, err
 			}
 
 			match = rule.Match(s)
```

Comment:
Could we simplify this and remove the `glob` dependency altogether?
Something like:

```go
if dbScope == "" {
// handle empty
}

if dbScope[len(dbScope)-1] == "*" {
// match = HasPrefix on dbScope[0 : len(dbScope)-1]
} else {
// exact match
}
```

Reply:
If we have this kind of check 
```
strings.ContainsAny(prefix, "*?")
```
Would it be a good idea to create a function for it
```
func ValidateScope(scope string) bool {
   prefix := scope[:len(scope)-1]
   return strings.ContainsAny(prefix, "*?")
}
```
So we can reuse that function for access control api and provisioner?

Reply:
Done and done